source/fundamentals/builders/indexes.txt

```diff
@@ -11,3 +11,150 @@ Indexes Builders
+queries on indexes. 
+
+The :java-core-api:`Indexes <com/mongodb/client/model/Indexes.html>`
+class provides static factory methods for all the MongoDB Index key
+types. Each method returns a :ref:`Bson <bson>` instance, which you can
+pass to `createIndex() <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#createIndex(com.mongodb.client.ClientSession,org.bson.conversions.Bson)>`_. 
+
+.. tip::
+
+   For brevity, you may choose to import all methods of the
+   :java-core-api:`Indexes <com/mongodb/client/model/Indexes.html>`
+   class statically: 
+
+   .. code-block:: java
+
+      import static com.mongodb.client.model.Indexes.*;
+
+   The examples below assume this static import.
+
+.. _index-ascending:
+
+Ascending
+---------
+
+Pass the field name you want to index to the 
+:java-core-api:`ascending() <com/mongodb/client/model/Indexes.html#ascending(java.lang.String...)>`
+method to create an ascending index. An ascending index will sort the
+index key from smallest to largest.
+
+The following example specifies an ascending index key for the ``name`` field:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Indexes.java
+   :language: java
+   :dedent:
+   :start-after: begin ascendingIndex
+   :end-before: end ascendingIndex
+
+.. _index-descending:
+
+Descending
+----------
+
+Pass the field name you want to index to the 
+:java-core-api:`descending() <com/mongodb/client/model/Indexes.html#descending(java.lang.String...)>`
+method to create a descending index. A descending index will sort the
+index key from largest to smallest.
+
+The following example specifies a compound index key composed of
+descending index keys for the ``capacity`` field: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Indexes.java
+   :language: java
+   :dedent:
+   :start-after: begin descendingIndex
+   :end-before: end descendingIndex
+
+.. _index-compound-indexes:
+
+Compound Indexes
+----------------
+
+Pass the field name you want to index to the 
+:java-core-api:`compoundIndex() <com/mongodb/client/model/Indexes.html#compoundIndex(org.bson.conversions.Bson...)>`
+method to create a compound index. 
+
+The following example specifies a compound index key composed of
+descending index keys for the ``capacity`` and ``year`` field, followed
+by an ascending index key for the ``name`` field: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Indexes.java
+   :language: java
+   :dedent:
+   :start-after: begin compoundIndexExample
+   :end-before: end compoundIndexExample
+
+.. _index-text-indexes:
+
+Text Indexes
+------------
+
+Pass the field name you want to index to the 
+:java-core-api:`text() <com/mongodb/client/model/Indexes.html#text(java.lang.String)>`
+method to create a text index key. A text index will group the index key
+by the word you specify.
+
+The following example specifies a text index key for "Theaters":
```

Comment:
Let's use lowercase for field names to follow javascript naming conventions. There's technically no restriction, but we use the javascript/JSON camelCase convention throughout our documentation.